server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java

```diff
@@ -30,6 +30,7 @@
 import java.util.Iterator;
```

Comment:
Why do you think so? MailImpl contains a resource `message` that we should probably don't try to evaluate in equals/hashcode. So I would argue the equality should not be provided by MailImpl.

Reply:
It was just convenient to implement the "serializable" concern.

Agree, let's do a "field by field" equality